pkg/compute/executor.go

```diff
@@ -300,7 +302,7 @@ func (e *BaseExecutor) Run(ctx context.Context, state store.LocalExecutionState)
 	stopwatch := telemetry.Timer(ctx, jobDurationMilliseconds, state.Execution.Job.MetricAttributes()...)
 	topic := EventTopicExecutionRunning
 	defer func() {
-		if err != nil {
+		if err.Error() != executor.ErrAlreadyCancelled.Error() {
```

Comment:
better to check using `!errors.Is(err, executor.ErrAlreadyCancelled)`

Reply:
So the problem there, is that we return a new error, by creating one from the one we get from [result](https://github.com/bacalhau-project/bacalhau/blob/main/pkg/compute/executor.go#L355) As a consequence of which it does not match the error type which we have created.

So this is what is happening

ErrAlreadyCanceled (at docker layer) -> Translated to string message and added to `RunCommanResult` -> And a new error type created.

We don't have a mapping of any error messages to specific type at computer/executor.go layer.

Do you have any other idea we can use. ?